crates/task-impls/src/quorum_proposal.rs

```diff
@@ -700,10 +588,13 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> QuorumProposalTaskState<TYPE
                 _fee,
             ) => {
                 let view = *view;
-                debug!(
-                    "Got payload commitment {:?} for view {view:?}",
-                    payload_commitment
-                );
+
+                if self.quorum_membership.get_leader(view) == self.public_key {
```

Comment:
Shouldn't we skip the dependency creation if we are not the leader?

Reply:
Wouldn't this statement imply that we are?